fdbclient/NativeAPI.actor.cpp

```diff
@@ -731,6 +731,13 @@ void DatabaseContext::setOption( FDBDatabaseOptions::Option option, Optional<Str
 			server_interf.clear();
 			locationCache.insert( allKeys, Reference<LocationInfo>() );
 			break;
+		case FDBDatabaseOptions::USE_PROVISIONAL_PROXIES:
```

Comment:
If this is called before `getMasterProxies` for a given `clientInfo`, then I think we will overwrite `masterProxies` again in the call to `getMasterProxies`. Maybe that's not technically wrong, but should we also set `masterProxiesLastChange` here?

Reply:
Also, do we need to do something with `masterProxies` in the `DATACENTER_ID` and `MACHINE_ID` options above?

Reply:
After looking at it, leaving masterProxiesLastChange to the current clientId appears to give us the correct behavior

Reply:
good catch, those need to check for provisional

Reply:
I think it does technically work if masterProxiesLastChange isn't updated, but it seems like it would be more within expectation to update it here. If you don't, I think it may end up doing a needless replacement of the list.